### DISCUSSION TITLE
Controlling a SuperFlo VS directly with GPIO instead of relays - configuration questions

### DISCUSSION THREAD

**Discussion by EgonSpengler** [tagyoureit/nodejs-poolController#960] [REM] [answered]
Right now I'm set up with four circuits, which I named Speed1..Speed3 and QuickClean, and connected to GPIO 22-25.  These have been added as Pump Circuits, with the corresponding GPIO pins also linked under Pump Relays.

My main problem: the circuits are not treated as mutually exclusive - selecting one speed should disable the others, but they all stay where toggled manually (I assume my triggers in REM are to blame here?)

Side question: Is there a way to relinquish manual control to allow the schedule to resume or re-trigger?  Since I currently have to toggle the scheduled speed off in order to enable another, it will stay off even after I turn off the rest.

Hopefully I've simply missed something obvious.  Thanks for all your work on this!

![REM](https://github.com/tagyoureit/nodejs-poolController/assets/4382519/ecb10b6d-4294-42dd-95a3-8e0ebd0c9370)
![circuits](https://github.com/tagyoureit/nodejs-poolController/assets/4382519/0233c6df-c3f2-45f8-ae5b-fbe00241dc3f)
![Circuits](https://github.com/tagyoureit/nodejs-poolController/assets/4382519/dddf7086-061e-4cf4-a434-be354878e80b)
![Relays](https://github.com/tagyoureit/nodejs-poolController/assets/4382519/6969ed01-0665-4ce9-86a6-be2566b30c9c)

**Comment by tagyoureit** [comment]
Don’t connect the aux circuits to the relays.  The circuits turn on the pump speeds which trigger the relays.  Connecting both to the same relay will cause problems.

**Comment by EgonSpengler** [comment]
After removing the aux circuit connections, I'm still able to toggle on any number of pump circuits simultaneously - the others will blink off for just a moment per their triggers, but stay on again.

I'm able to get what I need with 4 circuit groups using Off/Ignore for all but the desired speed, but this feels like a workaround as I'm now ignoring the Pumps tile on the dashboard except as a status monitor.  It works, though, so I'll proceed as it is.

**Reply by tagyoureit** [reply]
Can you attach a [replay](https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution#-dashpanel) and we'll take a look?

**Reply by tagyoureit** [reply, marked as answer]
> My main problem: the circuits are not treated as mutually exclusive - selecting one speed should disable the others, but they all stay where toggled manually (I assume my triggers in REM are to blame here?)

I just took a quick look at our logic and the pump manuals.   The pump can accept multiple, simultaneous inputs and it will default to the highest priority.  There is no expectation by the pump or our code for mutual exclusivity.  

Per the manual:
> Note: If more than one input (switch) is present, then the pump will give priority to the highest number input.  Therefore OVERRIDE has highest priority followed by STEP 3, then STEP 2, then STEP 1.

In your setup, the priority at the pump side would be QuickClean -> Speed3 -> Speed2 -> Speed1.

I would remove all of your triggers on the REM side and try to work with this logic.  njsPC will have all the schedules, logic, etc you need to control the pump and the triggers on the REM side will interfere with this.

**Reply by EgonSpengler** [reply]
Thanks for the help!  I finally got back from vacation and got (mostly) everything working, including schedules.  Removing the REM triggers and aux circuit connections did the trick.

My only last curiosity is that the scheduler asks for a "heat source" only when Speed 1 is selected -- I don't have a heater, so I tried both "No Change" as well as creating a dummy heater with no connection, but schedules using Speed 1 won't trigger.  Since I only use two speeds anyway, I won't be digging any further as it may be a configuration problem on my end.
